pkg/controllers/job/job_controller_handler.go

```diff
@@ -212,9 +212,14 @@ func (cc *Controller) updatePod(oldObj, newObj interface{}) {
 	}
 
 	event := vkbatchv1.OutOfSyncEvent
+	var exitCode int32
 	if oldPod.Status.Phase != v1.PodFailed &&
 		newPod.Status.Phase == v1.PodFailed {
 		event = vkbatchv1.PodFailedEvent
+		// TODO: currently only one container pod is supported by volcano
+		// Once multi containers pod is supported, update accordingly.
+		exitCode = newPod.Status.ContainerStatuses[0].State.Terminated.ExitCode
```

Comment:
let's open an issue to trace that. one option is to set something like "1:1" which means considering first container's exist code (or by container name, e.g. worker:-2)

Reply:
the default value is 0, the "first" container.

Reply:
I will open an issue following this.